extension/ts/main/extensions/tags/validation.ts

```diff
@@ -0,0 +1,141 @@
+import {getAllTags, getAncestry} from "../../adapters/tags";
+import {createModal} from "../../ui/modal";
+import {loaderOverlaid} from "../../ui/loadingIndicator";
+import {isAuthorized} from "../author/util/isAuthorized";
+import {UIColour} from "../../ui/colour";
+import {OnSave} from "../../entities/bookForm";
+
+declare const SPREADSHEET_ID: string; // set by webpack
+
+type GetTagsOptions = {noCache: boolean};
+
+const createHighlighterComponents = () => {
+	const highlighter = document.createElement("div");
+	highlighter.className = "bookEditInput"; // this comes from LibraryThing. Stealing their css
+	highlighter.id = "vbl-text-highlighter";
+
+	const backdrop = document.createElement("div");
+	backdrop.id = "vbl-text-backdrop";
+	backdrop.append(highlighter);
+
+	return {highlighter, backdrop};
+};
+
+const applyHighlights = async (text: string): Promise<string> => {
+	const validTags = await getAllTags();
+	const futureParts = text.split(",").map(async (part) => {
+		const trimmedPart = part.trim();
+		if (validTags.has(trimmedPart)) {
+			return part;
+		} else {
+			const mark = document.createElement("mark");
```

Comment:
i think this currently marks author tags (which get suggested by librarything) as invalid tags which could be confusing

Reply:
I think this is not a problem with our implementation, but a problem with the completeness of the Tag Index. user error!